### PR TITLE
like, comment, report

### PR DESCRIPTION
**Pull request by NguyenCongViet**
ready

### CODE REVIEW

File: app/Http/Controllers/User/LikeController.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $user = Auth::id();
+
+        $like = Like::where('user_id', $user)->where('post_id', $id)->first();
+
+        if (!$like) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/LikeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/User/LikeController.php</br>**Line**: 80</br>Expected 1 space between "'user_id'" and double arrow; 3 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/LikeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterDoubleArrow</br>**Severity**: error</br>**File**: app/Http/Controllers/User/LikeController.php</br>**Line**: 80</br>Expected 1 space between double arrow and "$request"; 2 found

File: app/Http/Controllers/User/LikeController.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            $like = Like::create([
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/LikeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/User/LikeController.php</br>**Line**: 81</br>Expected 1 space between "'type'" and double arrow; 6 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/LikeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterDoubleArrow</br>**Severity**: error</br>**File**: app/Http/Controllers/User/LikeController.php</br>**Line**: 81</br>Expected 1 space between double arrow and "1"; 2 found

File: app/Http/Controllers/User/LikeController.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                'user_id'   =>  $request->get('user'),
+                'type'      =>  1,
+                'post_id' => $request->get('post_id'),
+            ]);
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/LikeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.ControlStructureSpacing.SpaceBeforeCloseBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/User/LikeController.php</br>**Line**: 85</br>Expected 0 spaces before closing bracket; 1 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/LikeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/User/LikeController.php</br>**Line**: 85</br>Expected 1 space(s) after closing parenthesis; found 0

File: app/Http/Controllers/User/LikeController.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                ]);
+            } else {
+                $like->update([
+                    'type' => 1,
+                ]);
+            }
+        }
+
+        return response()->json([
+            'error' => false,
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/LikeController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Controllers/User/LikeController.php</br>**Line**: 99</br>Each line in an array declaration must end in a comma

File: app/Http/Controllers/User/ReportController.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $user = Auth::id();
+
+        $report = Report::where('user_id', $user)->where('post_id', $id)->first();
+
+        if (!$report) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/ReportController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReportController.php</br>**Line**: 80</br>Expected 1 space between "'user_id'" and double arrow; 3 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/ReportController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterDoubleArrow</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReportController.php</br>**Line**: 80</br>Expected 1 space between double arrow and "$request"; 2 found

File: app/Http/Controllers/User/ReportController.php [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            $report = Report::create([
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/ReportController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterComma</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReportController.php</br>**Line**: 81</br>Expected 1 space between "'type'" and double arrow; 6 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/ReportController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.SpaceAfterDoubleArrow</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReportController.php</br>**Line**: 81</br>Expected 1 space between double arrow and "1"; 2 found

File: app/Http/Controllers/User/ReportController.php [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+                'user_id'   =>  $request->get('user'),
+                'type'      =>  1,
+                'post_id' => $request->get('post_id'),
+            ]);
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/ReportController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.ControlStructureSpacing.SpaceBeforeCloseBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReportController.php</br>**Line**: 85</br>Expected 0 spaces before closing bracket; 1 found

Comment (framgiaciv3) [inline app/Http/Controllers/User/ReportController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReportController.php</br>**Line**: 85</br>Expected 1 space(s) after closing parenthesis; found 0

File: app/Http/Controllers/User/ReportController.php [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                ]);
+            } else {
+                $report->update([
+                    'type' => 1,
+                ]);
+            }
+        }
+
+        return response()->json([
+            'error' => false,
```
Comment (framgiaciv3) [inline app/Http/Controllers/User/ReportController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Controllers/User/ReportController.php</br>**Line**: 99</br>Each line in an array declaration must end in a comma

File: app/Http/Controllers/User/CommentController.php [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        ]);
+
+        $comment = Comment::where('post_id', $request->post_id)
+            ->where('user_id', $user->id)
+            ->orderBy('id', 'desc')
+            ->first();
+
+        return response()->json([
+            'error' => false,
+            'message' => 'Thêm comment thành công!',
```
Comment (BunnyPi04) [inline app/Http/Controllers/User/CommentController.php]: i18n

File: app/Http/Controllers/User/LikeController.php [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $user = Auth::id();
+        $like = Like::where('user_id', $user)->where('post_id', $id)->first();
```
Comment (BunnyPi04) [inline app/Http/Controllers/User/LikeController.php]: viết vài repository đi

File: app/Http/Controllers/User/CommentController.php [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $comment = Comment::findOrFail($id);
+        $comment->delete();
+
+        return response()->json([
+            'error' => false,
+            'message' => 'Delete Success',
```
Comment (BunnyPi04) [inline app/Http/Controllers/User/CommentController.php]: i18n

File: app/Http/Controllers/User/LikeController.php [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+                $like->update([
+                    'user_id' => $request->user_id,
+                    'type' => 0,
+                ]);
+            }
+        }
+
+        return response()->json([
+            'error' => false,
+            'message' => 'Like Success',
```
Comment (BunnyPi04) [inline app/Http/Controllers/User/LikeController.php]: i18n

File: resources/js/user.js [hunk 13]
```diff
@@ -43,3 +43,218 @@ $('#title').on('keyup', function () {
...
+    $.ajaxSetup({
+        headers: {'X-CSRF-Token': $('meta[name=_token]').attr('content')}
+    });
+
+    $.ajax({
+        type: 'POST',
+        url: route('comments.store'),
+        data: formdata,
+        success: function (res) {
+            console.log(res);
```
Comment (BunnyPi04) [inline resources/js/user.js]: xóa console.log

File: resources/js/user.js [hunk 14]
```diff
@@ -43,3 +43,218 @@ $('#title').on('keyup', function () {
...
+            if (res.info.images == null) {
+                $avatar = '/img/avatar-5.png';
```
Comment (BunnyPi04) [inline resources/js/user.js]: config đâu?

File: resources/js/user.js [hunk 15]
```diff
@@ -43,3 +43,218 @@ $('#title').on('keyup', function () {
...
+                headers: {
+                    'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+                }
+            });
+
+            $.ajax({
+                type: 'DELETE',
+                url: 'cfs/comments/' + comment_id,
+                success: function (res) {
+                    console.log(res);
```
Comment (BunnyPi04) [inline resources/js/user.js]: xóa console.log

**Pull request merged**